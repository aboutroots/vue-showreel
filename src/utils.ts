export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
