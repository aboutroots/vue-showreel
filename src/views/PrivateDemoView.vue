<template>
  <section class="PrivateDemoView container">
    <p class="copy">Welcome to secret page, for logged in users only!</p>
    <img alt="Gem image" class="gem-img float" src="@/assets/img/gem.png" />
    <img src="@/assets/img/secure.svg" class="bg-img" alt="lock image" />
  </section>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'PrivateDemoView',
  head: {
    title: {
      inner: 'Secret page',
    },
  },
});
</script>
<style lang="scss" scoped>
.container {
  display: flex;
  justify-content: center;
  flex-direction: column;
}
.gem-img {
  max-width: 300px;
  margin-top: 1rem;
  margin: 0 auto;
}
.copy {
  text-align: center;
  line-height: 1.5;
  max-width: 800px;
  margin: 0 auto;
}
</style>
