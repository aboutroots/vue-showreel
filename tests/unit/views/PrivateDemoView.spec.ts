import PrivateDemoView from '@/views/PrivateDemoView.vue';
import { shallowMount } from '../utils';

describe('PrivateDemoView tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(PrivateDemoView);

    const container = wrapper.find('.PrivateDemoView');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(PrivateDemoView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
