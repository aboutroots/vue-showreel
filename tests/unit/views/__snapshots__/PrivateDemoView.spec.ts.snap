// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateDemoView tests suite matches snapshot 1`] = `
<section
  class="PrivateDemoView container"
>
  <p
    class="copy"
  >
    Welcome to secret page, for logged in users only!
  </p>
   
  <img
    alt="Gem image"
    class="gem-img float"
    src="@/assets/img/gem.png"
  />
   
  <img
    alt="lock image"
    class="bg-img"
    src="@/assets/img/secure.svg"
  />
</section>
`;
